theano/gpuarray/opt.py

```diff
@@ -461,29 +741,32 @@ def max_inputs_to_GpuElemwise(node):
 
 @register_opt('fast_compile')
 @op_lifter([tensor.DimShuffle])
-def local_gpua_dimshuffle(node, context_name):
-    return GpuDimShuffle(node.op.input_broadcastable,
-                         node.op.new_order)
+@register_opt2([tensor.DimShuffle], 'fast_compile')
+def local_gpua_dimshuffle(op, context_name, inputs, outputs):
+    return GpuDimShuffle(op.input_broadcastable,
+                         op.new_order)
 
 
 @register_opt('fast_compile')
 @op_lifter([tensor.SpecifyShape])
-def local_gpua_specifyShape(node, context_name):
-    if isinstance(node.inputs[0].type, GpuArrayType):
+@register_opt2([tensor.SpecifyShape], 'fast_compile')
+def local_gpua_specifyShape(op, context_name, inputs, outputs):
+    if isinstance(inputs[0].type, GpuArrayType):
         return
-    inp = [as_gpuarray_variable(node.inputs[0], context_name)]
-    inp += node.inputs[1:]
+    inp = [as_gpuarray_variable(inputs[0], context_name)]
+    inp += inputs[1:]
     return tensor.specify_shape(*inp)
 
 
 @register_opt('fast_compile')
 @op_lifter([theano.compile.ops.Shape])
-def local_gpua_shape(node, context_name):
+@register_opt2([tensor.compile.ops.Shape], 'fast_compile')
+def local_gpua_shape(op, context_name, inputs, outputs):
     # op_lifter will call this opt too frequently as the output is
     # always on the CPU.
-    if isinstance(node.inputs[0].type, GpuArrayType):
+    if isinstance(inputs[0].type, GpuArrayType):
         return
```

Comment:
Yes. It makes sense to me. I was discussing the exact of what you said with @nouiz . I will do them